File: src/StackAndQueue/ImplementStackQueues.java
package StackAndQueue;

import java.util.LinkedList;
import java.util.Queue;

// https://leetcode.com/problems/implement-stack-using-queues/
public class ImplementStackQueues {
    final private Queue<Integer> myStack;
    public ImplementStackQueues() {
        this.myStack = new LinkedList<>();
    }

    public void push(int x) {
        Queue<Integer> temp = new LinkedList<>();
        while (!myStack.isEmpty()) {
            temp.add(myStack.poll());
        }
        myStack.add(x);

        while (!temp.isEmpty()) {
            myStack.add(temp.poll());
        }
    }

    public int pop() {
        return this.myStack.poll();
    }

    public int top() {
        return this.myStack.peek();
    }

    public boolean empty() {
        return this.myStack.isEmpty();
    }

    public static void main(String[] args) {
        ImplementStackQueues i = new ImplementStackQueues();
        i.push(1);
        i.push(2);
        i.push(3);
        i.push(4);
        System.out.println(i.top());
        i.pop();
        System.out.println(i.top());
        i.push(5);
        System.out.println(i.top());
    }
}

File: py/essential/ImplementStackQueues.py
# https://leetcode.com/problems/implement-stack-using-queues/description/
# Neet


class MyStack:

    def __init__(self):
      self.queue = []

    def push(self, x: int) -> None:
        tempQ = self.queue.copy()
        self.queue.clear()
        self.queue.append(x)
        self.queue.extend(tempQ)
        print(self.queue)
      

    def pop(self) -> int:
        if len(self.queue) == 0:
            return None
        return self.queue.pop(0)


    def top(self) -> int:
        return self.queue[0]

    def empty(self) -> bool:
        return len(self.queue) == 0
  
s=MyStack()
s.push(1)
s.push(2)
s.push(3)
print(s.pop())
print(s.pop())
print(s.pop())